integration-tests/gradle/src/integrationTest/kotlin/org/jetbrains/dokka/it/gradle/TestedVersions.kt

```diff
@@ -2,28 +2,32 @@ package org.jetbrains.dokka.it.gradle
 
 internal object TestedVersions {
 
+    // Kotlin and Gradle compatibility matrix: https://kotlinlang.org/docs/gradle.html
     val BASE =
         BuildVersions.permutations(
-            gradleVersions = listOf("7.4.2", "6.9"),
+            gradleVersions = listOf("7.5.1", "6.7.1"),
             kotlinVersions = listOf("1.7.20", "1.6.21", "1.5.31", "1.4.32"),
         ) + BuildVersions.permutations(
-            gradleVersions = listOf(*ifExhaustive("7.0", "6.1.1")),
-            kotlinVersions = listOf(*ifExhaustive("1.6.0", "1.5.0", "1.4.0"))
+            gradleVersions = listOf(*ifExhaustive("7.0", "6.7.1")),
+            kotlinVersions = listOf(*ifExhaustive("1.7.0", "1.6.0", "1.5.0", "1.4.0"))
         )
 
-    /**
-     * Starting with version 7, major android gradle plugin versions are aligned
-     * with major gradle versions, i.e android plugin 7.0.0 will not work with gradle 6.9
-     */
+    // Starting with version 7, major android gradle plugin versions are aligned
+    // with major gradle versions, i.e android plugin 7.0.0 will not work with gradle 6.9.
+    // Android and Gradle plugins compatibility matrix: https://developer.android.com/studio/releases/gradle-plugin
     val ANDROID =
         BuildVersions.permutations(
-            gradleVersions = listOf("7.4.2", *ifExhaustive("7.0")),
+            androidGradlePluginVersions = listOf("7.3.0"),
+            gradleVersions = listOf("7.5.1"),
             kotlinVersions = listOf("1.7.20", "1.6.21", "1.5.31", "1.4.32"),
-            androidGradlePluginVersions = listOf("7.2.0")
         ) + BuildVersions.permutations(
-            gradleVersions = listOf("6.9", *ifExhaustive("6.1.1", "5.6.4")),
+            androidGradlePluginVersions = listOf("4.2.0"),
+            gradleVersions = listOf("6.7.1"),
             kotlinVersions = listOf("1.7.20", "1.6.21", "1.5.31", "1.4.32"),
-            androidGradlePluginVersions = listOf("4.0.0", *ifExhaustive("3.6.3"))
+        ) + BuildVersions.permutations(
+            androidGradlePluginVersions = listOf(*ifExhaustive("3.6.4")),
+            gradleVersions = listOf(*ifExhaustive("6.7.1")),
+            kotlinVersions = listOf(*ifExhaustive("1.7.20", "1.6.21", "1.5.31", "1.4.32")),
```

Comment:
Made some changes which should hopefully be compatible. This may help repair `isExhaustive` flagged runs.

1. Latest AGP with latest Gradle and supported Kotlin versions. Version 7 of AGP works with Gradle 7 only
2. Previous major AGP version with its supported Gradle version (AGP 4.2+ works with Gradle 6.7)
3. Really old version of AGP, not sure if it'll work